### PR TITLE
GLSP-813 Use Node example server instead of java variant

### PR DESCRIPTION
**Pull request by tortmayr**
## Use Node example server instead of java variant 

Remove scripts to download Java Example Server and consume node variant via NPM instead Update README to reflect the changes.
Part of https://github.com/eclipse-glsp/glsp/issues/813

## Add support for launching WebSocket GLSP Servers 

Refactor socket launch classes to also support Websocket based connections & adapt example code & launch configs. 
(Pendant to https://github.com/eclipse-glsp/glsp/issues/984)

## Add support for direct server usage without an exernal process 

Introduce new `NodeGLSPVscodeServer` that directly connects to a GLSP server object instead of communicating via external process 
Fixes https://github.com/eclipse-glsp/glsp/issues/975

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -275,50 +297,80 @@
     rimraf "^3.0.2"
     typescript "^4.9.3"
 
-"@eclipse-glsp/dev@next":
-  version "1.1.0-next.7026c40.129"
-  resolved "https://registry.yarnpkg.com/@eclipse-glsp/dev/-/dev-1.1.0-next.7026c40.129.tgz#2ab7e736c1594e8a26556a34abbad6f43524e48c"
-  integrity sha512-VwR4xEPbEM5ZrX+WHj1iuoLfIVSmfH7iMteD/sy1AdzaL5DbytTASblVsUlO/VH6FGu8Au8EVLWVepBz7GG45Q==
+"@eclipse-glsp/dev@^1.1.0-next.acae6b0.132":
```
Comment (martin-fleck-at) [inline yarn.lock:300]: When building my yarn.lock file gets updated here, could you please double check that the latest version is checked in?

**Review by martin-fleck-at — CHANGES_REQUESTED** [review]
Very cool work, thank you very much Tobias! I tested all the use cases with the workflow example launch configurations:
- socket (local and external)
- websocket (local and external)
- integrated

Everything seems to work really well, really great!

As usual, I have a few comments and questions that could be addressed. The most important are, however, the compilation error (minor adaptation in the example) and the double checking of the regular expression that CodeQL is complaining about.

File: CHANGELOG.MD [hunk 2]
```diff
@@ -4,7 +4,17 @@
...
+
+### Breaking Changes
+
+-   [deps] Update to vscode-jsonrpc 8.0.2 & update minimum requirements for Node to >=16.11.0 [#31](https://github.com/eclipse-glsp/glsp-vscode-integration/pull/31)
+-   [launch] Refactor socket-based quickstart components to also support WebSocket connections [#37](https://github.com/eclipse-glsp/glsp-vscode-integration/pull/37)
+    -   Renamed `JavaSocketServerLaunchOptions` -> `SocketServerLaunchOptions` and removed `serverType` property
+    -   Renamed `GlspServerLauncher` -> `GLSPSocketServerLauncher`
+    -   Replaced `serverPort` property of `SocketGlspVscodeServerOptions` with `connectionOptions`
+    -   Added `start()` and `onReady` to `GlspVscodeServer` interface
+    -   Introduced new `NodeGlspVscodeServer`
```
[the remark this answers is not in the document]
Comment (tortmayr) [reply CHANGELOG.MD:17]: It's not, but I'm not sure what the best practice for handling PRs that contain both breaking and non-breaking changes would be. I could also create a separate entry in the non-breaking section. But two entries for a single PR are also kind of unexpected, right?

[the remark this answers is not in the document]
Comment (martin-fleck-at) [reply CHANGELOG.MD:17]: I definitely can argue for both cases. I think from a user perspective having the same PR appear multiple times seems a bit more logical than having the changes mixed but that is just my opinion and especially in this case it is very obvious that something new is not breaking so we can leave it for now.

File: .vscode/launch.json [hunk 3]
```diff
@@ -14,11 +14,12 @@
         "--extensionDevelopmentPath=${workspaceFolder}/example/workflow/extension"
       ],
       "outFiles": [
-        "${workspaceFolder}/example/workflow/extension/lib/*.js",
+        "${workspaceFolder}/example/workflow/extension/lib/**/*.js",
         "${workspaceFolder}/packages/vscode-integration/lib/**/*.js"
       ],
       "sourceMaps": true
     },
+
```
[the remark this answers is not in the document]
Comment (martin-fleck-at) [reply .vscode/launch.json:22]: It's okay to leave it as it does not do any harm.

**Pull request merged**